code/modules/projectiles/projectile.dm

```diff
@@ -357,8 +357,12 @@
 	if((speed * world.tick_lag) >= get_dist(current_turf, target_turf))
 		SEND_SIGNAL(src, COMSIG_BULLET_TERMINAL)
 
+
+	var/obj/item/hardpoint/hardpoint = shot_from
+	var/list/ignore_list = istype(hardpoint) ? list(hardpoint.owner) : null //if fired from a vehicle, exclude the vehicle's body from the adjacency check
```

Comment:
shouldn't `null` be an empty list instead?

Reply:
It doesn't have to be. Everything works just fine with `ignore_list = null`. I figured it was preferable to only initialize if needed. I'll rewrite it to make it clearer.